TAOK3.py

```diff
@@ -0,0 +1,376 @@
+
+    Return:
+    - file: file with fastq extension containing selected fragments.
+
+    For more information please see README
+
+    """
+    if type(length_bounds) is int:
+        length_bounds = 0, length_bounds
+    if type(gc_bounds) is int or type(gc_bounds) is float:
+        gc_bounds = 0, gc_bounds
+    sequences = SeqIO.parse(input_path, "fastq")
+    good_reads = []
+    for seq_record in sequences:
+        if (is_in_gc_bounds(seq_record, gc_bounds) and
+            is_in_length_bounds(seq_record, length_bounds) and
+            is_above_quality_threshold(seq_record, quality_threshold)):
+            good_reads += [seq_record]
+    if len(good_reads) == 0:
+        raise ValueError('There are no sequences suited to requirements')
+    if output_filename is None:
+        input_filename = os.path.split(input_path)[-1]
+        output_filename = input_filename
+    if not (output_filename.endswith('.fastq')):
+        output_filename = output_filename + '.fastq'
+    current_directory = os.getcwd()
+    path = os.path.join(current_directory, 'fastq_filtrator_results')
+    if not (os.path.exists(path)):
+        os.mkdir(path)
+    output_path = os.path.join(path, output_filename)
+    if os.path.exists(output_path):
+        error = 'File with such name exists! Change output_filename arg!'
+        raise ValueError(error)
+    SeqIO.write(good_reads, output_path, "fastq")
+
+
+class BiologicalSequence(ABC):
+    """
+    The abstract class for biological sequences
+    """
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, item):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    def __repr__(self):
+        pass
+
+    @abstractmethod
+    def is_alphabet_correct(self):
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Class for nucleic acids
+    """
+    def __init__(self, seq):
+        self.seq = seq
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __getitem__(self, item):
+        return self.seq[item]
+
+    def __str__(self):
+        return self.seq
+
+    def __repr__(self):
+        return self.seq
+
+    def is_alphabet_correct(self) -> bool:
+        """
+        The function check does the sequence contain
+        standard A, G, C, T, U nucleotides
+
+        Returns: bool - the result of check
+        """
+        return type(self).is_correct(self)
+
+    def complement(self) -> BiologicalSequence:
+        """
+        Output the complementary sequence
+        The complementarity rule could be found here:
+        https://en.wikipedia.org/wiki/Complementarity_(molecular_biology)
+
+        Arguments:
+        - self: the sequence obj to change
+
+        Return: DNAsequence or RNAsequence - the result sequence object
+        """
+        complement = self.seq.translate(type(self).rule_complement)
+        complement_seq = type(self)(complement)
```

Comment:
Можно как альтернативу использовать: self.\_\_class__(complement) (две пары кавычек немного смущают, но это дело вкуса)
```suggestion
        complement_seq = self.__class__(complement)
```